preprocess.lua

```diff
@@ -1,37 +1,35 @@
 require('onmt.init')
 
-local tds = require('tds')
-
-local cmd = torch.CmdLine()
-
-cmd:text("")
-cmd:text("preprocess.lua")
-cmd:text("")
-cmd:text("**Preprocess Options**")
-cmd:text("")
-cmd:text("")
-cmd:option('-config', '', [[Read options from this file]])
-
-cmd:option('-train_src', '', [[Path to the training source data]])
-cmd:option('-train_tgt', '', [[Path to the training target data]])
-cmd:option('-valid_src', '', [[Path to the validation source data]])
-cmd:option('-valid_tgt', '', [[Path to the validation target data]])
-
-cmd:option('-save_data', '', [[Output file for the prepared data]])
-
-cmd:option('-src_vocab_size', '50000', [[Comma-separated list of source vocabularies size: word[,feat1,feat2,...].]])
-cmd:option('-tgt_vocab_size', '50000', [[Comma-separated list of target vocabularies size: word[,feat1,feat2,...].]])
-cmd:option('-src_vocab', '', [[Path to an existing source vocabulary]])
-cmd:option('-tgt_vocab', '', [[Path to an existing target vocabulary]])
-cmd:option('-features_vocabs_prefix', '', [[Path prefix to existing features vocabularies]])
-
-cmd:option('-src_seq_length', 50, [[Maximum source sequence length]])
-cmd:option('-tgt_seq_length', 50, [[Maximum target sequence length]])
-cmd:option('-shuffle', 1, [[Shuffle data]])
-cmd:option('-seed', 3435, [[Random seed]])
-
-cmd:option('-report_every', 100000, [[Report status every this many sentences]])
+local cmd = onmt.ExtendedCmdLine.new("preprocess.lua")
+
+-- first argument define the dataType: BITEXT/MONO - default is BITEXT
+local dataType = 'BITEXT'
```

Comment:
Not a fan of these uppercase values. I prefer `bitext` and `mono` (for consistency, isn't `monotext` more appropriate?).

Also for training: `seq2seq` and `lm`.